src/js/components/DateInput/utils.js

```diff
@@ -125,6 +125,30 @@ const pullDigits = (text, index) => {
   return text.slice(index, end);
 };
 
+export const validateDateBounds = ({ dateBounds, selectedDate }) => {
+  if (!dateBounds || !selectedDate) return selectedDate;
+
+  const [startDate, endDate] = dateBounds.map((date) =>
+    new Date(date).toISOString(),
+  );
+
+  const selectedDateArray = Array.isArray(selectedDate)
+    ? selectedDate
+    : [selectedDate];
+
+  const isoSelectedDateArray = selectedDateArray.map((date) =>
+    new Date(date).toISOString(),
+  );
+
+  const allDatesOk = isoSelectedDateArray.every(
```

Comment:
```suggestion
  const validDate = isoSelectedDateArray.every(
```
```suggestion
  const validSelection = isoSelectedDateArray.every(
```
Small naming suggestions

Reply:
done